Add Permission Boundary Support to AWS Load Balancer Controller

bobdoah opened the pull request:
### What does this PR do?

Allow a permissions boundary to be applied to the IRSA IAM role. Additionally, support applying it to the role created by the AWS load balancer controller addon.  

### Motivation

We use a IAM role to provision our Terraform which has permissions boundary attached. That policy must be applied to all IAM roles created by the Terraform. It is used to prevent privilege escalation in keeping with https://aws.amazon.com/premiumsupport/knowledge-center/iam-permission-boundaries/
 

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR


### For Moderators
- [ ] E2E Test successfully complete before merge?


### Additional Notes

I would be happy to add the same functionality to the other addons, as they're adopted by us. I do not have the capacity to test them at the moment.

vara-bonthu commented on modules/kubernetes-addons/aws-load-balancer-controller/README.md:
```diff
@@ -58,6 +58,7 @@ Here is the link to get the AWS ELB [service annotations](https://kubernetes-sig
 
 
 <!--- BEGIN_TF_DOCS --->
+<!--- END_TF_DOCS --->
```
See the above comment

vara-bonthu commented on modules/kubernetes-addons/aws-load-balancer-controller/README.md:
```diff
@@ -116,5 +118,3 @@ No requirements.
 | <a name="output_argocd_gitops_config"></a> [argocd\_gitops\_config](#output\_argocd\_gitops\_config) | Configuration used for managing the add-on with ArgoCD |
 | <a name="output_ingress_name"></a> [ingress\_name](#output\_ingress\_name) | AWS LoadBalancer Controller Ingress Name |
 | <a name="output_ingress_namespace"></a> [ingress\_namespace](#output\_ingress\_namespace) | AWS LoadBalancer Controller Ingress Namespace |
-
-<!--- END_TF_DOCS --->
```
This shouldn't be removed

vara-bonthu reviewed (changes requested):
Hi @bobdoah Thanks for your first contribution on our repo 👍🏼 
I understand this is useful feature to control the IAM roles. 

PR Looks good to me. I have one comment on docs that needs addressing

kcoleman731 commented:
@bobdoah thank you for the PR! 

To aid in adding support to other add-ons, it probably makes sense to add permission boundary support the [`helm-addon` submodule](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/modules/kubernetes-addons/helm-addon). LBC add-on doesn't leverage the `helm-addon` submodule yet, but our plan is to update all add-ons which deploy helm charts to leverage it in the near future. We could prioritize updating the LBC add-on if you need.

bobdoah commented:
Thanks @vara-bonthu, @kcoleman731 for the prompt response.

I have corrected the documentation markers, an oversight on my part. I had thought I had inserted the docs in the correct place.

@kcoleman731 I have added support to the `helm-addon` module as suggested. It didn't look like there were many modules using that, at the moment. It would be great if you could prioritise the LBC change. My intent was to revisit the other modules as I adopted them.

bobdoah commented:
Realised I need to add this to the modules using `helm-addon` or they will not work.

vara-bonthu reviewed (approved):
Nice Work on this PR 👍🏼  LGTM!

bobdoah commented:
Any chance of getting this merged? Is there something I need to do fix the pr-test action?
